### PR TITLE
Work on px4io firmware building.

### PR DESCRIPTION
**Pull request by jgoppert**
This gets pretty far on building the px4io firmware, it is commented out for now until I figure out the link error.

```bash
Scanning dependencies of target firmware_io_nuttx
[ 91%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/adc.c.obj
[ 91%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/controls.c.obj
[ 92%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/dsm.c.obj
[ 92%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/px4io.c.obj
[ 92%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/registers.c.obj
[ 93%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/safety.c.obj
[ 93%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/sbus.c.obj
[ 93%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/systemlib/up_cxxinitialize.c.obj
[ 94%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/systemlib/perf_counter.c.obj
[ 94%] Building CXX object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/mixer.cpp.obj
[ 94%] Building CXX object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/systemlib/mixer/mixer.cpp.obj
[ 95%] Building CXX object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/systemlib/mixer/mixer_group.cpp.obj
[ 95%] Building CXX object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/systemlib/mixer/mixer_multirotor.cpp.obj
[ 96%] Building CXX object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/systemlib/mixer/mixer_simple.cpp.obj
[ 96%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/systemlib/pwm_limit/pwm_limit.c.obj
[ 96%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/__/lib/rc/st24.c.obj
[ 97%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/__/lib/rc/sumd.c.obj
[ 97%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/__/drivers/stm32/drv_hrt.c.obj
[ 97%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/__/drivers/stm32/drv_pwm_servo.c.obj
[ 98%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/__/drivers/boards/px4io-v2/px4iov2_init.c.obj
[ 98%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/__/drivers/boards/px4io-v2/px4iov2_pwm_servo.c.obj
[ 98%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/serial.c.obj
[100%] Building C object src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/__/systemlib/hx_stream.c.obj
Linking CXX executable firmware_io_nuttx
/home/jgoppert/gcc-arm-none-eabi-4_8-2014q3/bin/../lib/gcc/arm-none-eabi/4.8.4/../../../../arm-none-eabi/lib/armv7-m/libm.a(lib_a-wf_sqrt.o): In function `sqrtf':
wf_sqrt.c:(.text.sqrtf+0x80): undefined reference to `__errno'
wf_sqrt.c:(.text.sqrtf+0x8a): undefined reference to `__errno'
collect2: error: ld returned 1 exit status
make[3]: *** [src/modules/px4iofirmware/firmware_io_nuttx] Error 1
make[2]: *** [src/modules/px4iofirmware/CMakeFiles/firmware_io_nuttx.dir/all] Error 2
make[1]: *** [all] Error 2
```

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Do you have -lnosys enabled? The other issue could be that you're linking against the wrong math lib (IO has no hardware floating point unit, so it needs an entirely different compiler profile compared to FMU)

**Comment by jgoppert** [comment]
@LorenzMeier  I do have nosys enabled. I also switched to the m3 math library. Still same link error, but I did manage to cleanup the build quite a bit. I think this is worth merging now, I got it building again on posix/ nuttx. Hopefully didn't break qurt either.

File: cmake/configs/nuttx_px4fmu-v2_default.cmake [hunk 1]
```diff
@@ -169,6 +169,18 @@ set(config_extra_builtin_cmds
 	sercon
 	)
 
+set(config_io_board
+	px4io-v2
+	)
+
+set(config_extra_libs
+	${CMAKE_SOURCE_DIR}/src/lib/mathlib/CMSIS/libarm_cortexM4lf_math.a
```
Comment (LorenzMeier) [inline cmake/configs/nuttx_px4fmu-v2_default.cmake]: we are not using CMSIS any more. I recommend to strip those.

**Comment by LorenzMeier** [comment]
If you run V=1 make in the old build system you should see all flags of all commands. Doing the same in CMake should then allow you to easily identify where things go wrong between the two builds.

**Comment by davids5** [comment]
It sounds to me as if the lib passes nas used are not correct.  Have a look in the ld files. 

```

		/*
		 * This is a hack to make the newlib libm __errno() call
		 * use the NuttX get_errno_ptr() function.
		 */
		__errno = get_errno_ptr;
```

Is not in pio.

And I suspect now something is being linked that is not the same as the make based build.

We have to make sure non nuttx libs arr not linked and that the math lib is one that can come from the tool chain.

**Comment by LorenzMeier** [comment]
Yes, that was along the lines of what I thought. Currently the MS5611 and HMC5883 drivers are crashing on startup, also indicating potential inconsistencies. A token-by-token comparison of the full compiler and linker calls is necessary.

**Comment by davids5** [comment]
I would compare the map files or nm output sorted. In theory all the symbols should be the same...

**Comment by LorenzMeier** [comment]
@jgoppert I only want to merge further CMake cleanups when we have CI coverage. The ripples are enormous and of each even minor refactoring, and we've all sunk already significant time. I want to focus on just fixing what's broken and otherwise have as little moving parts as possible. When we have CI coverage for all platforms and OS we can get more experimental again. Right now its just a recipe to wear out the team quickly.

**Comment by LorenzMeier** [comment]
@jgoppert For an example of what I mean see this comment:
https://github.com/PX4/Firmware/pull/2836#issuecomment-139800553

I was in the same place this morning. Nothing worked and I had to sink 30 mins of debugging to just get what I left yesterday building again.

**Comment by jgoppert** [comment]
@davids5, @LorenzMeier  Used meld on the sorted nm output to look at the diff. there are definetly some differences, but nothing that is a smoking gun to me. Here is a comparison of the compilation:

```bash
arm-none-eabi-g++   -DCONFIG_ARCH_BOARD_PX4FMU_V2 -DPX4_MAIN=ms5611_app_main -D__PX4_NUTTX -fno-exceptions -fno-rtti -std=gnu++0x -fno-threadsafe-statics -DCONFIG_WCHAR_BUILTIN -D__CUSTOM_FILE_IO__ -Wall -Wno-sign-compare -Wextra -Wfloat-equal -Wframe-larger-than=1024 -Wpointer-arith -Wmissing-declarations -Wpacked -Wno-unused-parameter -Werror=format-security -Werror=array-bounds -Wfatal-errors -Werror=unused-variable -Werror=reorder -Werror=uninitialized -Werror=init-self -Wno-unused-const-variable -Werror=unused-but-set-variable -Wformat=1 -Wdouble-promotion -Werror=double-promotion -Wno-missing-field-initializers -Os -fno-strict-aliasing -fomit-frame-pointer -funsafe-math-optimizations -ffunction-sections -fdata-sections -fno-strength-reduce -fno-builtin-printf -fvisibility=hidden -include /home/jgoppert/git/px4/cmake-Firmware/src/include/visibility.h -mcpu=cortex-m4 -mthumb -march=armv7e-m -mfpu=fpv4-sp-d16 -mfloat-abi=hard -nodefaultlibs -nostdlib -I/home/jgoppert/git/px4/cmake-Firmware/src -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src -I/home/jgoppert/git/px4/cmake-Firmware/src/modules -I/home/jgoppert/git/px4/cmake-Firmware/src/include -I/home/jgoppert/git/px4/cmake-Firmware/src/lib -I/home/jgoppert/git/px4/cmake-Firmware/src/platforms -I/home/jgoppert/git/px4/cmake-Firmware/src/drivers/boards/px4fmu-v2 -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src/modules/px4_messages -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src/modules -I/home/jgoppert/git/px4/cmake-Firmware/mavlink/include/mavlink -I/home/jgoppert/git/px4/cmake-Firmware/src/lib/eigen -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/include -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/include/cxx -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/arch/chip -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/arch/common    -Os -o CMakeFiles/drivers__ms5611.dir/ms5611_spi.cpp.obj -c /home/jgoppert/git/px4/cmake-Firmware/src/drivers/ms5611/ms5611_spi.cpp
arm-none-eabi-g++   -DCONFIG_ARCH_BOARD_PX4FMU_V2 -DPX4_MAIN=ms5611_app_main -D__PX4_NUTTX -fno-exceptions -fno-rtti -std=gnu++0x -fno-threadsafe-statics -DCONFIG_WCHAR_BUILTIN -D__CUSTOM_FILE_IO__ -Wall -Wno-sign-compare -Wextra -Wfloat-equal -Wframe-larger-than=1024 -Wpointer-arith -Wmissing-declarations -Wpacked -Wno-unused-parameter -Werror=format-security -Werror=array-bounds -Wfatal-errors -Werror=unused-variable -Werror=reorder -Werror=uninitialized -Werror=init-self -Wno-unused-const-variable -Werror=unused-but-set-variable -Wformat=1 -Wdouble-promotion -Werror=double-promotion -Wno-missing-field-initializers -Os -fno-strict-aliasing -fomit-frame-pointer -funsafe-math-optimizations -ffunction-sections -fdata-sections -fno-strength-reduce -fno-builtin-printf -fvisibility=hidden -include /home/jgoppert/git/px4/cmake-Firmware/src/include/visibility.h -mcpu=cortex-m4 -mthumb -march=armv7e-m -mfpu=fpv4-sp-d16 -mfloat-abi=hard -nodefaultlibs -nostdlib -I/home/jgoppert/git/px4/cmake-Firmware/src -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src -I/home/jgoppert/git/px4/cmake-Firmware/src/modules -I/home/jgoppert/git/px4/cmake-Firmware/src/include -I/home/jgoppert/git/px4/cmake-Firmware/src/lib -I/home/jgoppert/git/px4/cmake-Firmware/src/platforms -I/home/jgoppert/git/px4/cmake-Firmware/src/drivers/boards/px4fmu-v2 -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src/modules/px4_messages -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src/modules -I/home/jgoppert/git/px4/cmake-Firmware/mavlink/include/mavlink -I/home/jgoppert/git/px4/cmake-Firmware/src/lib/eigen -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/include -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/include/cxx -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/arch/chip -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/arch/common    -Os -o CMakeFiles/drivers__ms5611.dir/ms5611_i2c.cpp.obj -c /home/jgoppert/git/px4/cmake-Firmware/src/drivers/ms5611/ms5611_i2c.cpp
arm-none-eabi-g++   -DCONFIG_ARCH_BOARD_PX4FMU_V2 -DPX4_MAIN=ms5611_app_main -D__PX4_NUTTX -fno-exceptions -fno-rtti -std=gnu++0x -fno-threadsafe-statics -DCONFIG_WCHAR_BUILTIN -D__CUSTOM_FILE_IO__ -Wall -Wno-sign-compare -Wextra -Wfloat-equal -Wframe-larger-than=1024 -Wpointer-arith -Wmissing-declarations -Wpacked -Wno-unused-parameter -Werror=format-security -Werror=array-bounds -Wfatal-errors -Werror=unused-variable -Werror=reorder -Werror=uninitialized -Werror=init-self -Wno-unused-const-variable -Werror=unused-but-set-variable -Wformat=1 -Wdouble-promotion -Werror=double-promotion -Wno-missing-field-initializers -Os -fno-strict-aliasing -fomit-frame-pointer -funsafe-math-optimizations -ffunction-sections -fdata-sections -fno-strength-reduce -fno-builtin-printf -fvisibility=hidden -include /home/jgoppert/git/px4/cmake-Firmware/src/include/visibility.h -mcpu=cortex-m4 -mthumb -march=armv7e-m -mfpu=fpv4-sp-d16 -mfloat-abi=hard -nodefaultlibs -nostdlib -I/home/jgoppert/git/px4/cmake-Firmware/src -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src -I/home/jgoppert/git/px4/cmake-Firmware/src/modules -I/home/jgoppert/git/px4/cmake-Firmware/src/include -I/home/jgoppert/git/px4/cmake-Firmware/src/lib -I/home/jgoppert/git/px4/cmake-Firmware/src/platforms -I/home/jgoppert/git/px4/cmake-Firmware/src/drivers/boards/px4fmu-v2 -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src/modules/px4_messages -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src/modules -I/home/jgoppert/git/px4/cmake-Firmware/mavlink/include/mavlink -I/home/jgoppert/git/px4/cmake-Firmware/src/lib/eigen -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/include -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/include/cxx -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/arch/chip -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/arch/common    -Os -o CMakeFiles/drivers__ms5611.dir/ms5611_nuttx.cpp.obj -c /home/jgoppert/git/px4/cmake-Firmware/src/drivers/ms5611/ms5611_nuttx.cpp
arm-none-eabi-ar cq libdrivers__ms5611.a  CMakeFiles/drivers__ms5611.dir/ms5611_spi.cpp.obj CMakeFiles/drivers__ms5611.dir/ms5611_i2c.cpp.obj CMakeFiles/drivers__ms5611.dir/ms5611_nuttx.cpp.obj
arm-none-eabi-ranlib libdrivers__ms5611.a
```

```bash
arm-none-eabi-g++ -MD -c -fno-exceptions -fno-rtti -std=gnu++0x -fno-threadsafe-statics -D__CUSTOM_FILE_IO__ -Wall -Wextra -Werror -Wdouble-promotion -Wshadow -Wfloat-equal -Wpointer-arith -Wmissing-declarations -Wpacked -Wno-unused-parameter -Werror=format-security -Werror=array-bounds -Wfatal-errors -Wformat=1 -Werror=unused-but-set-variable -Werror=unused-variable -Werror=double-promotion -Werror=reorder -Werror=uninitialized -Werror=init-self -Wno-missing-field-initializers -Os -g3 -fno-strict-aliasing -fno-strength-reduce -fomit-frame-pointer -funsafe-math-optimizations -fno-builtin-printf -ffunction-sections -fdata-sections -mcpu=cortex-m4 -mthumb -march=armv7e-m -mfpu=fpv4-sp-d16 -mfloat-abi=hard   -DCONFIG_ARCH_BOARD_PX4FMU_V2 -D__PX4_NUTTX -DCONFIG_WCHAR_BUILTIN   -I/home/jgoppert/git/px4/Firmware/src/ -I/home/jgoppert/git/px4/Firmware/src//modules/ -I/home/jgoppert/git/px4/Firmware/src/include/ -I/home/jgoppert/git/px4/Firmware/src/lib/ -I/home/jgoppert/git/px4/Firmware/src/platforms/ -I/home/jgoppert/git/px4/Firmware/Build/versioning/ -I/home/jgoppert/git/px4/Firmware/src/drivers/boards/px4fmu-v2 -I/home/jgoppert/git/px4/Firmware/src/lib/eigen/ -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/include -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/include/cxx -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/arch/chip -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/arch/common -DPX4_MAIN=ms5611_app_main -fvisibility=hidden -include /home/jgoppert/git/px4/Firmware/src/include/visibility.h /home/jgoppert/git/px4/Firmware/src/drivers/ms5611/ms5611_nuttx.cpp -o ms5611_nuttx.cpp.o
arm-none-eabi-g++ -MD -c -fno-exceptions -fno-rtti -std=gnu++0x -fno-threadsafe-statics -D__CUSTOM_FILE_IO__ -Wall -Wextra -Werror -Wdouble-promotion -Wshadow -Wfloat-equal -Wpointer-arith -Wmissing-declarations -Wpacked -Wno-unused-parameter -Werror=format-security -Werror=array-bounds -Wfatal-errors -Wformat=1 -Werror=unused-but-set-variable -Werror=unused-variable -Werror=double-promotion -Werror=reorder -Werror=uninitialized -Werror=init-self -Wno-missing-field-initializers -Os -g3 -fno-strict-aliasing -fno-strength-reduce -fomit-frame-pointer -funsafe-math-optimizations -fno-builtin-printf -ffunction-sections -fdata-sections -mcpu=cortex-m4 -mthumb -march=armv7e-m -mfpu=fpv4-sp-d16 -mfloat-abi=hard   -DCONFIG_ARCH_BOARD_PX4FMU_V2 -D__PX4_NUTTX -DCONFIG_WCHAR_BUILTIN   -I/home/jgoppert/git/px4/Firmware/src/ -I/home/jgoppert/git/px4/Firmware/src//modules/ -I/home/jgoppert/git/px4/Firmware/src/include/ -I/home/jgoppert/git/px4/Firmware/src/lib/ -I/home/jgoppert/git/px4/Firmware/src/platforms/ -I/home/jgoppert/git/px4/Firmware/Build/versioning/ -I/home/jgoppert/git/px4/Firmware/src/drivers/boards/px4fmu-v2 -I/home/jgoppert/git/px4/Firmware/src/lib/eigen/ -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/include -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/include/cxx -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/arch/chip -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/arch/common -DPX4_MAIN=ms5611_app_main -fvisibility=hidden -include /home/jgoppert/git/px4/Firmware/src/include/visibility.h /home/jgoppert/git/px4/Firmware/src/drivers/ms5611/ms5611_spi.cpp -o ms5611_spi.cpp.o
arm-none-eabi-g++ -MD -c -fno-exceptions -fno-rtti -std=gnu++0x -fno-threadsafe-statics -D__CUSTOM_FILE_IO__ -Wall -Wextra -Werror -Wdouble-promotion -Wshadow -Wfloat-equal -Wpointer-arith -Wmissing-declarations -Wpacked -Wno-unused-parameter -Werror=format-security -Werror=array-bounds -Wfatal-errors -Wformat=1 -Werror=unused-but-set-variable -Werror=unused-variable -Werror=double-promotion -Werror=reorder -Werror=uninitialized -Werror=init-self -Wno-missing-field-initializers -Os -g3 -fno-strict-aliasing -fno-strength-reduce -fomit-frame-pointer -funsafe-math-optimizations -fno-builtin-printf -ffunction-sections -fdata-sections -mcpu=cortex-m4 -mthumb -march=armv7e-m -mfpu=fpv4-sp-d16 -mfloat-abi=hard   -DCONFIG_ARCH_BOARD_PX4FMU_V2 -D__PX4_NUTTX -DCONFIG_WCHAR_BUILTIN   -I/home/jgoppert/git/px4/Firmware/src/ -I/home/jgoppert/git/px4/Firmware/src//modules/ -I/home/jgoppert/git/px4/Firmware/src/include/ -I/home/jgoppert/git/px4/Firmware/src/lib/ -I/home/jgoppert/git/px4/Firmware/src/platforms/ -I/home/jgoppert/git/px4/Firmware/Build/versioning/ -I/home/jgoppert/git/px4/Firmware/src/drivers/boards/px4fmu-v2 -I/home/jgoppert/git/px4/Firmware/src/lib/eigen/ -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/include -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/include/cxx -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/arch/chip -I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/arch/common -DPX4_MAIN=ms5611_app_main -fvisibility=hidden -include /home/jgoppert/git/px4/Firmware/src/include/visibility.h /home/jgoppert/git/px4/Firmware/src/drivers/ms5611/ms5611_i2c.cpp -o ms5611_i2c.cpp.o
arm-none-eabi-ld -Ur -Map /home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build//home/jgoppert/git/px4/Firmware/src/drivers/ms5611/module.pre.o.map  -o /home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build//home/jgoppert/git/px4/Firmware/src/drivers/ms5611/module.pre.o ms5611_nuttx.cpp.o ms5611_spi.cpp.o ms5611_i2c.cpp.o
arm-none-eabi-objcopy --localize-hidden /home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build//home/jgoppert/git/px4/Firmware/src/drivers/ms5611/module.pre.o
```

**Comment by jgoppert** [comment]
That is hard to see, here is a version without the warning flags and line by line, without allowing any reordering.

cmake

```bash
arm-none-eabi-g++ 
	-DCONFIG_ARCH_BOARD_PX4FMU_V2
	-DPX4_MAIN=ms5611_app_main
	-D__PX4_NUTTX
	-fno-exceptions
	-fno-rtti
	-std=gnu++0x
	-fno-threadsafe-statics
	-DCONFIG_WCHAR_BUILTIN
	-D__CUSTOM_FILE_IO__
	-Os
	-fno-strict-aliasing
	-fomit-frame-pointer
	-funsafe-math-optimizations
	-ffunction-sections
	-fdata-sections
	-fno-strength-reduce
	-fno-builtin-printf
	-fvisibility=hidden
	-include /home/jgoppert/git/px4/cmake-Firmware/src/include/visibility.h
	-mcpu=cortex-m4
	-mthumb
	-march=armv7e-m
	-mfpu=fpv4-sp-d16
	-mfloat-abi=hard
	-nodefaultlibs
	-nostdlib
	-I/home/jgoppert/git/px4/cmake-Firmware/src
	-I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default
	-I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src
	-I/home/jgoppert/git/px4/cmake-Firmware/src/modules
	-I/home/jgoppert/git/px4/cmake-Firmware/src/include
	-I/home/jgoppert/git/px4/cmake-Firmware/src/lib
	-I/home/jgoppert/git/px4/cmake-Firmware/src/platforms
	-I/home/jgoppert/git/px4/cmake-Firmware/src/drivers/boards/px4fmu-v2
	-I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src/modules/px4_messages
	-I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/src/modules
	-I/home/jgoppert/git/px4/cmake-Firmware/mavlink/include/mavlink
	-I/home/jgoppert/git/px4/cmake-Firmware/src/lib/eigen
	-I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/include -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/include/cxx -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/arch/chip -I/home/jgoppert/git/px4/cmake-Firmware/build_px4fmu-v2_default/px4fmu-v2/NuttX/nuttx-export/arch/common    -Os -o CMakeFiles/drivers__ms5611.dir/ms5611_spi.cpp.obj -c /home/jgoppert/git/px4/cmake-Firmware/src/drivers/ms5611/ms5611_spi.cpp
arm-none-eabi-ar
	cq libdrivers__ms5611.a
	CMakeFiles/drivers__ms5611.dir/ms5611_spi.cpp.obj
	CMakeFiles/drivers__ms5611.dir/ms5611_i2c.cpp.obj
	CMakeFiles/drivers__ms5611.dir/ms5611_nuttx.cpp.obj
arm-none-eabi-ranlib libdrivers__ms5611.a
```

make
```bash
arm-none-eabi-g++
	-MD
	-c 
	-fno-exceptions
	-fno-rtti -std=gnu++0x
	-fno-threadsafe-statics
	-D__CUSTOM_FILE_IO__
	-Os
	-g3
	-fno-strict-aliasing
	-fno-strength-reduce
	-fomit-frame-pointer
	-funsafe-math-optimizations
	-fno-builtin-printf
	-ffunction-sections
	-fdata-sections
	-mcpu=cortex-m4
	-mthumb -march=armv7e-m
	-mfpu=fpv4-sp-d16
	-mfloat-abi=hard
	-DCONFIG_ARCH_BOARD_PX4FMU_V2
	-D__PX4_NUTTX
	-DCONFIG_WCHAR_BUILTIN
	-I/home/jgoppert/git/px4/Firmware/src/
	-I/home/jgoppert/git/px4/Firmware/src//modules/
	-I/home/jgoppert/git/px4/Firmware/src/include/
	-I/home/jgoppert/git/px4/Firmware/src/lib/
	-I/home/jgoppert/git/px4/Firmware/src/platforms/
	-I/home/jgoppert/git/px4/Firmware/Build/versioning/
	-I/home/jgoppert/git/px4/Firmware/src/drivers/boards/px4fmu-v2
	-I/home/jgoppert/git/px4/Firmware/src/lib/eigen/
	-I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/include
	-I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/include/cxx
	-I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/arch/chip
	-I/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build/nuttx-export/arch/common
	-DPX4_MAIN=ms5611_app_main
	-fvisibility=hidden
	-include /home/jgoppert/git/px4/Firmware/src/include/visibility.h
	/home/jgoppert/git/px4/Firmware/src/drivers/ms5611/ms5611_nuttx.cpp
	-o
	ms5611_nuttx.cpp.o
arm-none-eabi-ld
	-Ur
	-Map
	/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build//home/jgoppert/git/px4/Firmware/src/drivers/ms5611/module.pre.o.map
	-o
	/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build//home/jgoppert/git/px4/Firmware/src/drivers/ms5611/module.pre.o
	ms5611_nuttx.cpp.o
	ms5611_spi.cpp.o
	ms5611_i2c.cpp.o
arm-none-eabi-objcopy
	--localize-hidden
	/home/jgoppert/git/px4/Firmware/Build/px4fmu-v2_default.build//home/jgoppert/git/px4/Firmware/src/drivers/ms5611/module.pre.o
```

**Comment by jgoppert** [comment]
After much headache, resorted diff version:

```bash
--- ms5611_build_cmake	2015-09-12 14:40:34.753689277 -0400
+++ ../Firmware/ms5611_build_orig	2015-09-12 14:40:39.901689126 -0400
@@ -1,48 +1,43 @@
 arm-none-eabi-g++
+	-MD
 	-c
 	-fno-exceptions
 	-fno-rtti
 	-std=gnu++0x
 	-fno-threadsafe-statics
 	-D__CUSTOM_FILE_IO__
 	-Os
+	-g3
 	-fno-strict-aliasing
 	-fno-strength-reduce
 	-fomit-frame-pointer
 	-funsafe-math-optimizations
 	-fno-builtin-printf
 	-ffunction-sections
 	-fdata-sections
 	-mcpu=cortex-m4
 	-mthumb
 	-march=armv7e-m
 	-mfpu=fpv4-sp-d16
 	-mfloat-abi=hard
 	-DCONFIG_ARCH_BOARD_PX4FMU_V2
 	-D__PX4_NUTTX
 	-DCONFIG_WCHAR_BUILTIN
 	-Isrc/
-	-Ibuild_px4fmu-v2_default
-	-Ibuild_px4fmu-v2_default/src
 	-Isrc/modules/
 	-Isrc/include/
 	-Isrc/lib/
 	-Isrc/platforms/
+	-IBuild/versioning/
 	-Isrc/drivers/boards/px4fmu-v2
-	-Isrc/modules/px4_messages
-	-Isrc/modules
-	-Imavlink/include/mavlink
 	-Isrc/lib/eigen/
 	-Inuttx-export/include
 	-Inuttx-export/include/cxx
 	-Inuttx-export/arch/chip
 	-Inuttx-export/arch/common
 	-DPX4_MAIN=ms5611_app_main
 	-fvisibility=hidden
 	-include src/include/visibility.h
-	-nodefaultlibs
-	-nostdlib
-	-Os
 	src/drivers/ms5611/ms5611_nuttx.cpp
 	-o
 	ms5611_nuttx.cpp.o
```

**Comment by LorenzMeier** [comment]
-nodefaultlibs
-nostdlib

Seems like it could be well needed.

**Comment by jgoppert** [comment]
@LorenzMeier You are looking at the diff backwards.

**Comment by jgoppert** [comment]
If nothing is obvious I can just try to change the flags to match. I remember having to add the nodefaultlibs and nostdlibs due to a link issue.

**Pull request closed without merging**